backend/telemetry_core/src/aggregator/aggregator_set.rs

```diff
@@ -19,12 +20,18 @@ impl AggregatorSet {
     /// Spawn the number of aggregators we're asked to.
     pub async fn spawn(
         num_aggregators: usize,
+        metadata_path: Option<PathBuf>,
         opts: AggregatorOpts,
     ) -> anyhow::Result<AggregatorSet> {
         assert_ne!(num_aggregators, 0, "You must have 1 or more aggregator");
 
         let aggregators = futures::future::try_join_all(
-            (0..num_aggregators).map(|_| Aggregator::spawn(opts.clone())),
+            std::iter::once(AggregatorOpts {
+                metadata_path,
+                ..opts.clone()
+            })
```

Comment:
Hm... does this mean multiple aggregators will concurrently write to disk? We don't want that, we should only have one that does writes.

Reply:
Nope, only the first one. This is why the first one has `metdata_path` set and the rest of aggregators have it as `None`.

Reply:
I see now, thanks. The old code with a simple iterator was much more readable for me, you could have had a simple `if` statement in there and leave the rest as is.